src/main.rs

```diff
@@ -196,26 +214,58 @@ fn main() {
         .cloned()
         .collect();
 
-    let mut cargo = Command::new(cargo_bin());
-
-    cargo.envs(std::env::vars_os());
-
-    cargo
-        .arg("test")
-        .args(first_args)
-        .arg("--")
-        .args(second_args);
+    // Build the final cargo test command.
+    let mut cargo = format!(
+        "{} {} {} -- {}",
+        cargo_bin(),
+        custom_cargo_command.as_deref().unwrap_or_else(|| "test"),
+        first_args
+            .iter()
+            .map(|x| x.to_string_lossy())
+            .fold(String::new(), |mut acc, f| {
+                acc.push(' ');
+                acc.push_str(&f);
+                acc
+            }),
+        second_args
+            .iter()
+            .map(|x| x.to_string_lossy())
+            .fold(String::new(), |mut acc, f| {
+                acc.push(' ');
+                acc.push_str(&f);
+                acc
+            }),
+    );
 
     // We do this weird shit here since cargo does not support running multiple
     // tests individually.
     for item in final_items {
-        cargo.args(["--skip", &item]);
+        cargo.push_str(" --skip ");
+        cargo.push_str(&item);
```

Comment:
Is there a specific reason of doing this string manipulation rather than building `Command` ?

Reply:
Honest answer is that someone smarter than me told me to avoid manipulation of opaque commands in a similar situation. But, I will attempt to illustrate an issue. Take the following as an example:

`cargo qtest -c "watch --env password='Super Secret' -- cargo test"`

If we try to parse, on whitespace for example, we get `vec![watch,--env,passoword='Super,Secret',--,cargo,test]`. We've lost the space between 'Super Secret', and created an extra arg, `Secret'`. 

Better to offload to `sh -c` which is known to handle it correctly (and hopefully `cmd /c` will also).

Reply:
It's far dangerous to manipulate strings as you can accidentally hit issues like https://github.com/onur-ozkan/cargo-qtest/pull/5#issuecomment-1913610020

I am quite sure we can handle subcommands with `Command` without ever needing to manipulate string commands.